src/main/java/org/molgenis/bbmri/eric/service/EricDownloadService.java

```
@@ -0,0 +1,117 @@
+
+	private static final String ERIC_SOURCE_ENTITY = BbmriEricPackage.NAME + '_' + EricSourceMetaData.ENTITY_NAME;
+
+	private DataService dataService;
+	private Gson gson;
+
+	@Autowired
+	public EricDownloadService(DataService dataService)
+	{
+		this.dataService = dataService;
+		this.gson = new Gson();
+	}
+
+	@Scheduled(cron = "0 0 0 * * *")
+	@RunAsSystem
+	@Transactional
+	public void downloadSources()
+	{
+		Iterable<Entity> it = dataService.findAll(ERIC_SOURCE_ENTITY);
+
+		if (!it.iterator().hasNext())
+		{
+			LOG.info("No ERIC sources to download.");
+			return;
+		}
+		else
+		{
+			LOG.info("Starting biobanks import of external ERIC sources.");
+		}
+
+		int adds = 0;
+		int updates = 0;
+		int sources = 0;
+		for (Entity source : it)
+		{
+			try
+			{
+				LOG.info(String.format("Importing ERIC biobanks from %s", source.get(EricSourceMetaData.SOURCE)
+						.toString()));
+
+				OutputStream out = new ByteArrayOutputStream();
+				URL request = new URL(source.get(EricSourceMetaData.SOURCE).toString());
+				FileCopyUtils.copy(request.openStream(), out);
+				BbmriEricDataResponse bedr = gson.fromJson(out.toString(), BbmriEricDataResponse.class);
+
+				for (Map<String, Object> biobank : bedr.getBiobanks())
+				{
+					DefaultEntity ericBiobank = new DefaultEntity(
+							dataService.getEntityMetaData(CatalogueMetaData.FULLY_QUALIFIED_NAME), dataService);
+					for (Entry<String, Object> entry : biobank.entrySet())
+					{
+						ericBiobank.set(entry.getKey(), entry.getValue());
+					}
+
+					// add/update
+					if (dataService.findOne(CatalogueMetaData.FULLY_QUALIFIED_NAME, ericBiobank.getIdValue()) == null)
+					{
+						dataService.add(CatalogueMetaData.FULLY_QUALIFIED_NAME, ericBiobank);
+						adds++;
+					}
+					else
+					{
+						dataService.update(CatalogueMetaData.FULLY_QUALIFIED_NAME, ericBiobank);
+						updates++;
+					}
+				}
+				sources++;
+			}
+			catch (Exception e)
```

catch(Throwable t) to catch all exceptions and errors